### PR TITLE
ISPN-5265 Optimize Getall operations over Hot Rod

### PR DESCRIPTION
**Pull request by wburns**
* Added remote cache getAll that invokes embedded version
* Changed embedded getAll to no longer return null values

https://issues.jboss.org/browse/ISPN-5266

### CODE REVIEW

**Comment by galderz** [comment]
Both the JIRA description and the commit message are misleading. This PR seems to be getAll rather than putAll?

**Comment by galderz** [comment]
Otherwise the rest looks good, great work @wburns :+1:

**Comment by wburns** [comment]
Whoops let me fix that.

**Comment by wburns** [comment]
Fixed commit message, thanks,

**Comment by galderz** [comment]
Integrated, thanks @wburns :)

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/ReplGetAllPerfTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   protected int numberOfHotRodServers() {
+      return 5;
+   }
+
+   @Override
+   protected ConfigurationBuilder clusterConfig() {
+      return hotRodCacheConfiguration(getDefaultClusteredCacheConfig(
+            CacheMode.DIST_SYNC, false));
+   }
+}
```
Comment (vchepeli) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/ReplGetAllPerfTest.java]: @wburns should not be here CacheMode.REPL_SYNC ?

Comment (wburns) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/stress/ReplGetAllPerfTest.java]: Yes

**Pull request closed without merging**